lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,533 @@
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
+  const files = testGranule1.files.concat(testGranule2.files);
+
+  await Promise.all(files.flatMap((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+
+  await Promise.all([
+    granulesModel.create(testGranule1),
+    granulesModel.create(testGranule2),
+  ]);
+
+  t.teardown(async () => {
+    granulesModel.delete({ granuleId: testGranule1.granuleId });
+    granulesModel.delete({ granuleId: testGranule2.granuleId });
+    await s3Utils.recursivelyDeleteS3Bucket(alternateBucket);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+    granulesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+  });
+  const records = await knex(tableNames.granules);
+  const fileRecords = await knex(tableNames.files);
+  t.is(records.length, 2);
+  t.is(fileRecords.length, 2);
+});
+
+test.serial('migrateGranulesAndFiles processes all non-failing granule records and does not process files of failling granule records', async (t) => {
+  const {
+    collectionPgModel,
+    executionPgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
+
+  const testExecution2 = fakeExecutionRecordFactory();
+  await executionPgModel.create(
+    knex,
+    testExecution2
+  );
+
+  const testCollection2 = fakeCollectionRecordFactory();
+  await collectionPgModel.create(
+    t.context.knex,
+    testCollection2
+  );
+
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
+  // remove required field so record will fail
+  delete testGranule.collectionId;
+
+  await Promise.all([
+    dynamodbDocClient().put({
+      TableName: process.env.GranulesTable,
+      Item: testGranule,
+    }).promise(),
+    granulesModel.create(testGranule2),
+  ]);
+
+  t.teardown(async () => {
+    granulesModel.delete({ granuleId: testGranule.granuleId });
+    granulesModel.delete({ granuleId: testGranule2.granuleId });
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 0,
+      success: 1,
+    },
+    granulesSummary: {
+      dynamoRecords: 2,
+      failed: 1,
+      skipped: 0,
+      success: 1,
+    },
+  });
+  const records = await knex(tableNames.granules);
+  const fileRecords = await knex(tableNames.files);
+  t.is(records.length, 1);
+  t.is(fileRecords.length, 1);
+});
+
+test.serial.skip('migrateFileRecord throws RecordAlreadyMigrated error for already migrated record', async () => {
+});
```

Comment:
Should we remove this one?

Reply:
I am unsure. https://github.com/nasa/cumulus/pull/2051#discussion_r564089098